plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -980,6 +984,15 @@ protected WriteMapping legacyToWriteMapping(@SuppressWarnings("unused") Connecto
                 .orElseGet(Function::identity);
     }
 
+    private Function<PreparedQuery, PreparedQuery> tryApplyOrderBy(Optional<List<SortItem>> sortOrder)
```

Comment:
This should take sort _and_ limit, and be named appropriately.

Reply:
I'm not convinced. We have already limit applying logic and we reuse limit field to pass topN count.

Reply:
I don't feel comfortable implementing infrastructure for ORDER BY  pushdown without proper testing.
Also, why TopN is equivalent to ORDER BY + LIMIT, the actual implementation may be different, or can depend on the value of `limit`.